torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -371,7 +372,13 @@ void TCPStore::waitForWorkers() {
       const auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(
           std::chrono::steady_clock::now() - start);
       if (timeout_ != kNoTimeout && elapsed > timeout_) {
-        break;
+        C10_THROW_ERROR(
+            DistStoreError,
+            fmt::format(
```

Comment:
I love fmt format as well. But someone trying to do this in https://github.com/pytorch/pytorch/pull/107268 had the PR reverted. I am not sure if this PR will have the same issue, I added the ci flag here.